pkg/cmd/pipelinerun/logs.go

```diff
@@ -72,134 +58,67 @@ func logCommand(p cli.Params) *cobra.Command {
 		Example:               eg,
 		Args:                  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			prName := args[0]
-
-			cs, err := p.Clients()
-			if err != nil {
-				return err
-			}
-
-			prLogs := NewPipelineRunLogs(prName, p.Namespace(), cs)
-
-			opt := LogOptions{
-				LogOptions: taskrun.LogOptions{
-					AllSteps: allSteps,
-				},
-				Tasks: onlyTasks,
-			}
-
-			s := logs.Streams{
-				Out: cmd.OutOrStderr(),
+			opts.pipelinerunName = args[0]
+			opts.stream = &logs.Stream{
+				Out: cmd.OutOrStdout(),
 				Err: cmd.OutOrStderr(),
 			}
 
-			//TODO: put log fetcher as pipelinerun dependency
-			prLogs.Fetch(s, opt, logs.DefaultLogFetcher(cs.Kube))
-
-			return nil
+			return opts.run()
 		},
 	}
 
-	c.Flags().BoolVarP(&allSteps, "all", "a", false, "show all logs including init steps injected by tekton")
-	c.Flags().StringSliceVarP(&onlyTasks, "only-tasks", "t", []string{}, "show the logs for mentioned task only")
+	c.Flags().BoolVarP(&opts.allSteps, "all", "a", false, "show all logs including init steps injected by tekton")
+	c.Flags().BoolVarP(&opts.follow, "follow", "f", false, "stream the live logs")
+	c.Flags().StringSliceVarP(&opts.tasks, "only-tasks", "t", []string{}, "show logs for mentioned tasks only")
 
 	return c
 }
 
-//NewPipelineRunLogs create a new instance of PipelineRunLogs.
-func NewPipelineRunLogs(pipelineRunName string, namespace string, clients *cli.Clients) *PipelineRunLogs {
-	return &PipelineRunLogs{
-		Name:    pipelineRunName,
-		Ns:      namespace,
-		Clients: clients,
+func (lo *LogOptions) run() error {
```

Comment:
:thinking: :bowing_man:

Reply:
@vdemeester I didn't find it particularly attractive as well (albeit I see that kubectl uses it). The problem we are trying to solve is to seperate out `how a command is run` from `how it is invoked` so that tests just need to construct the `Command Executor` (LogOptions in this case) , invoke `run` and need not rely on `cobra`

Reply:
This could be done with a method that doesn't take any cobra related struct (aka `LogOptions` and other params) ? The end result is the same, so all good to me anyway :wink: